client/llb/marshal.go

```diff
@@ -101,7 +101,7 @@ func MarshalConstraints(base, override *Constraints) (*pb.Op, *pb.OpMetadata) {
 			Architecture: c.Platform.Architecture,
 			Variant:      c.Platform.Variant,
 			OSVersion:    c.Platform.OSVersion,
-			OSFeatures:   c.Platform.OSFeatures,
+			OSFeatures:   append([]string{}, c.Platform.OSFeatures...),
```

Comment:
Ooops, broke a couple of tests because they expect `[]string(nil)` and are getting `[]string{}`. It's too late tonight, but I reckon I'll just change the test data. The two are basically equivalent in Go anyway.

Reply:
We should keep the `nil` value. It is not the same in JSON for example.

Reply:
Yeah, that makes sense. It's mildly annoying that `append(nil, nil...)` produces an empty slice, rather than propagating the `nil`, so I guess I don't get to have a one-liner here.

Reply:
Following up, even though I fixed this, [the field is marked  `omitempty`](https://github.com/opencontainers/image-spec/blob/v1.0/specs-go/v1/descriptor.go#L57-L59) for which the [Golang json/encoding marshaller will omit the field if nil _or_ `len() == 0`](https://pkg.go.dev/encoding/json#Marshal). So it turns out they _are_ the same in JSON, at least when serialising out from Go in this case. (Which has bitten us elsewhere in the past when reading Go-serialised JSON from other, stricter, languages, but the effective equivalence of `nil` and an empty slice is pretty deeply embedded in Go.)